### ISSUE TITLE
Streaming Desktop - Video appears pixelated

### ISSUE THREAD

**Issue by zthun** [LizardByte/Sunshine#2451]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

Did not install the nightly.

### Describe the Bug

Upgraded from 0.22.2 to 0.23.0.  Issue does not happen on 0.22.2. 

After about a minute of moving my mouse or navigating through Steam Big Picture, the video all of a sudden appears pixelated and corrupted.  There doesn't seem to be a specific trigger.  This happens when streaming to my steam deck, from my Apple TV, and my M2 Mac Mini using Moonlight.

![image](https://github.com/LizardByte/Sunshine/assets/2132881/34d1c698-e47a-4c49-88f4-4f49a29c2fa7)

### Expected Behavior

In 0.22.2, this wasn't happening and the stream was smooth.  I would expect the same behavior.

### Host Operating System

Windows

### Operating System Version

Windows 11 Pro 22H2

### Architecture

64 bit

### Sunshine commit or version

0.23.0

### Package

Windows - Chocolatey (Third Party)

### GPU Type

AMD

### GPU Model

RX Vega64

### GPU Driver/Mesa Version

24.3.1

### Config

```Shell
(empty)
```

### Apps

```json
{
    "env": "",
    "apps": [
        {
            "name": "Desktop",
            "image-path": "desktop.png"
        },
        {
            "name": "Steam Big Picture",
            "cmd": "steam:\/\/open\/bigpicture",
            "auto-detach": "true",
            "wait-all": "true",
            "image-path": "steam.png"
        },
        {
            "name": "MoonDeckStream",
            "output": "",
            "cmd": "C:\\Users\\zthun\\AppData\\Local\\Programs\\MoonDeckBuddy\\bin\\MoonDeckStream.exe",
            "exclude-global-prep-cmd": "false",
            "elevated": "false",
            "auto-detach": "false",
            "wait-all": "true",
            "exit-timeout": "5",
            "image-path": "steam-deck.png"
        }
    ]
}
```

### Relevant log output

```shell
[2024:04:20:16:50:36]: Info: Sunshine version: 0.23.0
[2024:04:20:16:50:36]: Info: nvprefs: NvAPI_Initialize() failed, ignore if you don't have NVIDIA video card
[2024:04:20:16:50:36]: Info: Compiling shaders...
[2024:04:20:16:50:37]: Info: System tray created
[2024:04:20:16:50:37]: Info: Compiled shaders
[2024:04:20:16:50:37]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024:04:20:16:50:37]: Info: Trying encoder [nvenc]
[2024:04:20:16:50:37]: Info: ddprobe.exe [1] [] returned: 0x00000000
[2024:04:20:16:50:37]: Info: Set GPU preference: 1
[2024:04:20:16:50:37]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:50:37]: Info: Active GPU has HAGS disabled
[2024:04:20:16:50:37]: Info: Using realtime GPU priority
[2024:04:20:16:50:37]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:50:37]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:50:37]: Info: Display refresh rate [60Hz]
[2024:04:20:16:50:37]: Info: Requested frame rate [60fps]
[2024:04:20:16:50:37]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:50:37]: Info: Encoder [nvenc] is not supported on this GPU
[2024:04:20:16:50:37]: Info: Trying encoder [quicksync]
[2024:04:20:16:50:37]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:50:37]: Info: Active GPU has HAGS disabled
[2024:04:20:16:50:37]: Info: Using realtime GPU priority
[2024:04:20:16:50:37]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:50:37]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:50:37]: Info: Display refresh rate [60Hz]
[2024:04:20:16:50:37]: Info: Requested frame rate [60fps]
[2024:04:20:16:50:37]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:50:37]: Info: Encoder [quicksync] is not supported on this GPU
[2024:04:20:16:50:37]: Info: Trying encoder [amdvce]
[2024:04:20:16:50:37]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:50:37]: Info: Active GPU has HAGS disabled
[2024:04:20:16:50:37]: Info: Using realtime GPU priority
[2024:04:20:16:50:37]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:50:37]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:50:37]: Info: Display refresh rate [60Hz]
[2024:04:20:16:50:37]: Info: Requested frame rate [60fps]
[2024:04:20:16:50:37]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:50:37]: Info: SDR color coding [Rec. 601]
[2024:04:20:16:50:37]: Info: Color depth: 8-bit
[2024:04:20:16:50:37]: Info: Color range: [JPEG]
[2024:04:20:16:50:37]: Error: [h264_amf @ 0000014d3addcc40] encoder->Init() failed with error 3
[2024:04:20:16:50:37]: Info: Retrying with fallback configuration options for [h264_amf] after error: Internal bug, should not have happened
[2024:04:20:16:50:37]: Info: SDR color coding [Rec. 601]
[2024:04:20:16:50:37]: Info: Color depth: 8-bit
[2024:04:20:16:50:37]: Info: Color range: [JPEG]
[2024:04:20:16:50:37]: Info: SDR color coding [Rec. 601]
[2024:04:20:16:50:37]: Info: Color depth: 8-bit
[2024:04:20:16:50:37]: Info: Color range: [JPEG]
[2024:04:20:16:50:37]: Error: [av1_amf @ 0000014d3addd400] CreateComponent(AMFVideoEncoderHW_AV1) failed with error 11
[2024:04:20:16:50:37]: Error: Could not open codec [av1_amf]: Encoder not found
[2024:04:20:16:50:38]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:50:38]: Info: Active GPU has HAGS disabled
[2024:04:20:16:50:38]: Info: Using realtime GPU priority
[2024:04:20:16:50:38]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:50:38]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:50:38]: Info: Display refresh rate [60Hz]
[2024:04:20:16:50:38]: Info: Requested frame rate [60fps]
[2024:04:20:16:50:38]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:50:38]: Info: SDR color coding [Rec. 709]
[2024:04:20:16:50:38]: Info: Color depth: 10-bit
[2024:04:20:16:50:38]: Info: Color range: [JPEG]
[2024:04:20:16:50:38]: Error: [hevc_amf @ 0000014d3added80] encoder->Init() failed with error 1
[2024:04:20:16:50:38]: Error: Could not open codec [hevc_amf]: Internal bug, should not have happened
[2024:04:20:16:50:38]: Info: 
[2024:04:20:16:50:38]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024:04:20:16:50:38]: Info: 
[2024:04:20:16:50:38]: Info: Found H.264 encoder: h264_amf [amdvce]
[2024:04:20:16:50:38]: Info: Found HEVC encoder: hevc_amf [amdvce]
[2024:04:20:16:50:38]: Info: Configuration UI available at [https://localhost:47990]
[2024:04:20:16:50:38]: Info: Registered Sunshine mDNS service
[2024:04:20:16:52:46]: Info: Encoder reenumeration is required
[2024:04:20:16:52:46]: Info: // Testing for available encoders, this may generate errors. You can safely ignore those errors. //
[2024:04:20:16:52:46]: Info: Trying encoder [nvenc]
[2024:04:20:16:52:46]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:52:46]: Info: Active GPU has HAGS disabled
[2024:04:20:16:52:46]: Info: Using realtime GPU priority
[2024:04:20:16:52:46]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:52:46]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:52:46]: Info: Display refresh rate [60Hz]
[2024:04:20:16:52:46]: Info: Requested frame rate [60fps]
[2024:04:20:16:52:46]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:52:46]: Info: Encoder [nvenc] is not supported on this GPU
[2024:04:20:16:52:46]: Info: Trying encoder [quicksync]
[2024:04:20:16:52:46]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:52:46]: Info: Active GPU has HAGS disabled
[2024:04:20:16:52:46]: Info: Using realtime GPU priority
[2024:04:20:16:52:46]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:52:46]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:52:46]: Info: Display refresh rate [60Hz]
[2024:04:20:16:52:46]: Info: Requested frame rate [60fps]
[2024:04:20:16:52:46]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:52:46]: Info: Encoder [quicksync] is not supported on this GPU
[2024:04:20:16:52:46]: Info: Trying encoder [amdvce]
[2024:04:20:16:52:46]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:52:46]: Info: Active GPU has HAGS disabled
[2024:04:20:16:52:46]: Info: Using realtime GPU priority
[2024:04:20:16:52:46]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:52:46]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:52:46]: Info: Display refresh rate [60Hz]
[2024:04:20:16:52:46]: Info: Requested frame rate [60fps]
[2024:04:20:16:52:46]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:52:46]: Info: SDR color coding [Rec. 601]
[2024:04:20:16:52:46]: Info: Color depth: 8-bit
[2024:04:20:16:52:46]: Info: Color range: [JPEG]
[2024:04:20:16:52:46]: Error: [h264_amf @ 0000014d3b56ee80] encoder->Init() failed with error 3
[2024:04:20:16:52:46]: Info: Retrying with fallback configuration options for [h264_amf] after error: Internal bug, should not have happened
[2024:04:20:16:52:46]: Info: SDR color coding [Rec. 601]
[2024:04:20:16:52:46]: Info: Color depth: 8-bit
[2024:04:20:16:52:46]: Info: Color range: [JPEG]
[2024:04:20:16:52:46]: Info: SDR color coding [Rec. 601]
[2024:04:20:16:52:46]: Info: Color depth: 8-bit
[2024:04:20:16:52:46]: Info: Color range: [JPEG]
[2024:04:20:16:52:46]: Error: [av1_amf @ 0000014d3b56e940] CreateComponent(AMFVideoEncoderHW_AV1) failed with error 11
[2024:04:20:16:52:46]: Error: Could not open codec [av1_amf]: Encoder not found
[2024:04:20:16:52:46]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:52:46]: Info: Active GPU has HAGS disabled
[2024:04:20:16:52:46]: Info: Using realtime GPU priority
[2024:04:20:16:52:46]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:52:46]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:52:46]: Info: Display refresh rate [60Hz]
[2024:04:20:16:52:46]: Info: Requested frame rate [60fps]
[2024:04:20:16:52:46]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:52:46]: Info: SDR color coding [Rec. 709]
[2024:04:20:16:52:46]: Info: Color depth: 10-bit
[2024:04:20:16:52:46]: Info: Color range: [JPEG]
[2024:04:20:16:52:46]: Error: [hevc_amf @ 0000014d3b56f040] encoder->Init() failed with error 1
[2024:04:20:16:52:46]: Error: Could not open codec [hevc_amf]: Internal bug, should not have happened
[2024:04:20:16:52:46]: Info: 
[2024:04:20:16:52:46]: Info: // Ignore any errors mentioned above, they are not relevant. //
[2024:04:20:16:52:46]: Info: 
[2024:04:20:16:52:46]: Info: Found H.264 encoder: h264_amf [amdvce]
[2024:04:20:16:52:46]: Info: Found HEVC encoder: hevc_amf [amdvce]
[2024:04:20:16:52:46]: Info: Executing [Desktop]
[2024:04:20:16:52:46]: Info: nvprefs: NvAPI_Initialize() failed, ignore if you don't have NVIDIA video card
[2024:04:20:16:52:46]: Info: CLIENT CONNECTED
[2024:04:20:16:52:47]: Info: 
Device Description : Radeon RX Vega
Device Vendor ID   : 0x00001002
Device Device ID   : 0x0000687F
Device Video Mem   : 8120 MiB
Device Sys Mem     : 0 MiB
Share Sys Mem      : 8155 MiB
Feature Level      : 0x0000B100
Capture size       : 2560x1080
Offset             : 0x0
Virtual Desktop    : 2560x1080
[2024:04:20:16:52:47]: Info: Active GPU has HAGS disabled
[2024:04:20:16:52:47]: Info: Using realtime GPU priority
[2024:04:20:16:52:47]: Info: Desktop resolution [2560x1080]
[2024:04:20:16:52:47]: Info: Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:52:47]: Info: Display refresh rate [60Hz]
[2024:04:20:16:52:47]: Info: Requested frame rate [60fps]
[2024:04:20:16:52:47]: Info: 
Colorspace         : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709
Bits Per Color     : 10
Red Primary        : [0.639648,0.339844]
Green Primary      : [0.299805,0.69043]
Blue Primary       : [0.137695,0.0380859]
White Point        : [0.282227,0.296875]
Min Luminance      : 0.5 nits
Max Luminance      : 270 nits
Max Full Luminance : 270 nits
[2024:04:20:16:52:47]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2024:04:20:16:52:47]: Info: SDR color coding [Rec. 601]
[2024:04:20:16:52:47]: Info: Color depth: 8-bit
[2024:04:20:16:52:47]: Info: Color range: [MPEG]
[2024:04:20:16:52:47]: Info: Gamepad 0 will be DualShock 4 controller (auto-selected by client-reported type)
[2024:04:20:16:52:56]: Info: Opening UI from system tray
[2024:04:20:16:52:56]: Info: Resolved user-provided command 'https://localhost:47990' to '"C:\Program Files\Mozilla Firefox\firefox.exe" -osint -url "https://localhost:47990"'
[2024:04:20:16:52:56]: Info: https://localhost:47990 running with PID 7560
[2024:04:20:16:52:56]: Info: Opened url [https://localhost:47990]
[2024:04:20:16:52:57]: Info: Web UI: [127.0.0.1] -- not authorized
```

**Comment by ReenigneArcher** [comment]
I think this was likely fixed by https://github.com/LizardByte/Sunshine/pull/2419

Did you actually try the nightly build?

**Comment by zthun** [comment]
@ReenigneArcher Went ahead and installed the nightly to test.  Issue seems to be fixed.  Saw version 0.23.1.  So far, not seeing the issues as severely as it was.

The only way I can reproduce this now is if I drag highlight a portion of the destkop, release, and then it pixelates for half a second before refreshing.   This should be good enough; this bug can be closed as fixed in 0.23.1.
